src/components/AddressField/AddressField.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react'
+import { useCallback, useEffect, useRef, useState } from 'react'
+import classNames from 'classnames'
+import Icon from 'semantic-ui-react/dist/commonjs/elements/Icon'
+import { InputOnChangeData } from 'semantic-ui-react/dist/commonjs/elements/Input'
+import Popup from 'semantic-ui-react/dist/commonjs/modules/Popup'
+import { Field } from '../Field/Field'
+import { shorten, isValid } from './utils'
+import { Props } from './AddressField.types'
+import './AddressField.css'
+
+export default function AddressField(props: Props) {
+  const {
+    className,
+    fieldClassName,
+    i18n,
+    resolveName,
+    onChange,
+    ...otherProps
+  } = props
+  const [inputValue, setInputValue] = useState('')
+  const [address, setAddress] = useState('')
+  const timeout = useRef<NodeJS.Timeout>()
+  const [valid, setValid] = useState<boolean>()
+  const [loading, setLaoding] = useState<boolean>()
+
+  useEffect(() => {
+    if (props.value && props.value !== address) {
+      setInputValue(props.value)
+    }
+  }, [props.value, address])
+
+  useEffect(() => {
+    return () => {
+      clearTimeout(timeout.current)
+    }
+  }, [])
+
+  const handleChange = useCallback(
+    (evt, data: InputOnChangeData) => {
+      setInputValue(data.value)
+      setValid(undefined)
+      setAddress('')
+      if (timeout.current) {
+        clearTimeout(timeout.current)
+      }
+
+      timeout.current = setTimeout(async () => {
+        if (isValid(data.value)) {
+          setValid(true)
+          if (onChange) {
+            onChange(evt, data)
+          }
+          return
+        }
+
+        setLaoding(true)
+        const resolvedAddress = await resolveName(data.value)
```

Comment:
Should we wrap this function into a `try ... catch`? If resolving a name fails due to an error it will keep its loading state, right? We might be able to show an error in the field message as well.